lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,84 @@
+return if ENV['skip_autoinject']
+
+begin
+  require 'open3'
+
+  _, status = Open3.capture2e({'skip_autoinject' => 'true'}, 'bundle show ddtrace')
+
+  if status.success?
+    puts '[DATADOG LIB INJECTION] ddtrace already installed... skipping auto-injection'
+    return
+  end
+
+  require 'bundler'
+  require 'shellwords'
+
+  if Bundler.frozen_bundle?
+    puts "[DATADOG LIB INJECTION] Cannot inject with frozen Bundler"
+    return
+  end
+
+  version = "<DD_TRACE_VERSION_TO_BE_REPLACED>"
+  sha = "<DD_TRACE_SHA_TO_BE_REPLACED>"
+
+  bundle_add_ddtrace_cmd =
+    if !version.empty?
+      # For public release
+      "bundle add ddtrace --require ddtrace/auto_instrument --version #{version.gsub(/^v/, '').shellescape}"
+    elsif !sha.empty?
+      # For internal testing
+      "bundle add ddtrace --require ddtrace/auto_instrument --github datadog/dd-trace-rb --ref #{sha.shellescape}"
+    end
+
+  unless bundle_add_ddtrace_cmd
+    puts "[DATADOG LIB INJECTION] NO VERSION"
+    return
+  end
```

Comment:
I see a number of error messages in this file. They seem like messages that we will show to customers, so I would suggest approaching them from the POV of a customer that knows nothing about dd-trace-rb.

For instance, compare getting

> [DATADOG LIB INJECTION] NO VERSION

and this example ([randomly picked from](https://github.com/DataDog/dd-trace-rb/pull/2646))

> W, [2023-02-24T10:33:59.671680 #772]  WARN -- ddtrace: [ddtrace] Profiling was requested but is not
supported, profiling disabled: Your ddtrace installation is missing support for the Continuous
Profiler because the `pkg-config` system tool is missing. This issue can usually be fixed by installing
one of the following: the `pkg-config` package on Homebrew and Debian/Ubuntu-based Linux; the `pkgconf`
package on Arch and Alpine-based Linux; the `pkgconf-pkg-config` package on Fedora/Red Hat-based
Linux. (Tip: When fixing this, ensure `pkg-config` is installed **before** running `bundle install`,
and remember to clear any installed gem caches). For help solving this issue, please contact Datadog
support at <https://docs.datadoghq.com/help/>. You can also check out the Continuous Profiler
troubleshooting page at <https://dtdg.co/ruby-profiler-troubleshooting>.

E.g. when writing this message, I think we should think about the customer that will see it. As an example, on this one in particular, I think this issue would probably be a packaging problem, that maybe the customer can't do anything about. So perhaps:

> [ddtrace] Datadog lib-injection failed: missing version configuration. We suggest trying to upgrade to the latest lib-injection version. For help solving this issue, please contact Datadog support at <https://docs.datadoghq.com/help/>.